lib/yoti/dynamic_share_service/policy/source_constraint.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module Yoti
+  module DynamicSharingService
+    # A list of anchors to require for a dynamic share
+    class SourceConstraint
+      DRIVING_LICENCE = 'DRIVING_LICENCE'
+      PASSPORT = 'PASSPORT'
+      NATIONAL_ID = 'NATIONAL_ID'
+      PASS_CARD = 'PASS_CARD'
+
+      SOURCE_CONSTRAINT = 'SOURCE'
+
+      attr_reader :anchors
+
+      def soft_preference
+        return @soft_preference if @soft_preference
+
+        false
+      end
+
+      def to_json(*_args)
+        as_json.to_json
+      end
+
+      def as_json
+        obj = {
+          type: SOURCE_CONSTRAINT,
+          preferred_sources: [
+            anchors: @anchors.as_json
+          ]
```

Comment:
`as_json` doesn't seem to work in the Sinatra example (I believe Rails ActiveModel adds support for this) - it also needs curly brackets.

This seems to fix the issue:
```ruby
preferred_sources: {
  anchors: @anchors.map {|anchor| anchor.as_json }
}
```

Perhaps we need some extra test coverage for adding constraints in `wanted_attribute_spec.rb`?